client/src/components/auth/SignUp.vue

```diff
@@ -0,0 +1,184 @@
+                    <label for="password_confirmation">Confirm password</label>
+                    <VTextField
+                        id="password_confirmation"
+                        :type="passConfirmVisible ? 'text' : 'password'"
+                        :append-icon="
+                            passConfirmVisible ? 'mdi-eye' : 'mdi-eye-off'
+                        "
+                        @click:append="passConfirmVisible = !passConfirmVisible"
+                        solo
+                        outlined
+                        dense
+                        v-model="registerData.password_confirmation"
+                        :rules="passwordConfirmationRules"
+                        ref="password_confirmation"
+                        :error-messages="passConfirmationError"
+                    ></VTextField>
+                </VCol>
+            </VForm>
+            <VBtn depressed large color="primary" class="btn" @click="onSignUp"
+                >Sign Up</VBtn
+            >
+        </div>
+        <Alert
+            :message="alert.message"
+            :visibility="alert.visible"
+            :type="alert.type"
+            @alert-closed="onAlertClose"
+        />
+    </div>
+</template>
+
+<script>
+import * as actions from '@/store/modules/auth/types/actions';
+import { mapActions } from 'vuex';
+import Alert from '@/components/alert/Alert';
+export default {
+    name: 'SignUp',
+    components: {
+        Alert
+    },
+    data: () => ({
+        formValid: false,
+        passVisible: false,
+        passConfirmVisible: false,
+        registerData: {
+            name: '',
+            email: '',
+            password: '',
+            password_confirmation: ''
+        },
+        nameRules: [
+            v => !!v || 'Name is required',
+            v => v.length >= 2 || 'Name must be more than 2 characters',
+            v => v.length <= 100 || 'Name must be less than 100 characters'
+        ],
+        emailRules: [
+            v => !!v || 'Email is required',
+            v => /.+@.+/.test(v) || 'Email must be valid',
+            v => v.length <= 50 || 'Email must be less than 50 characters'
+        ],
+        passwordRules: [
+            v => !!v || 'Password is required',
+            v => v.length >= 8 || 'Password must be min 8 characters length'
+        ],
+        passwordConfirmationRules: [
+            v => !!v || 'Password confirmation is required',
+            v => v.length >= 8 || 'Password must be min 8 characters length'
+        ],
+        alert: {
+            visible: false,
+            message: 'Error',
+            type: ''
+        }
+    }),
+    methods: {
+        ...mapActions('auth', {
+            signUp: actions.SIGN_UP
+        }),
+        async onSignUp() {
+            this.$refs.form.validate();
+            if (this.formValid) {
+                try {
+                    await this.signUp(this.registerData);
+                    this.alert.message =
```

Comment:
create method `showMessage(message, type = 'success')` and use it for both error and success